mesonbuild/modules/_qt.py

```diff
@@ -616,3 +734,364 @@ def compile_translations(self, state: 'ModuleState', args: T.Tuple, kwargs: 'Com
+        KwargInfo('include_directories', ContainerTypeInfo(list, (build.IncludeDirs, str)), listify=True, default=[]),
+        KwargInfo('namespace', str, default=''),
+        KwargInfo('typeinfo', str, default=''),
+
+        KwargInfo('moc_extra_args', ContainerTypeInfo(list, str), listify=True, default=[]),
+        KwargInfo('qmlcachegen_extra_args', ContainerTypeInfo(list, str), listify=True, default=[]),
+        KwargInfo('qmltyperegistrar_extra_args', ContainerTypeInfo(list, str), listify=True, default=[]),
+
+        KwargInfo('generate_qmldir', bool, default=True),
+        KwargInfo('generate_qmltype', bool, default=True),
+        KwargInfo('create_plugin_target', bool, default=True),
+        KwargInfo('generate_plugin_source', bool, default=True),
+        KwargInfo('cachegen', bool, default=True),
+
+        KwargInfo('dependencies', ContainerTypeInfo(list, (Dependency, ExternalLibrary)), listify=True, default=[]),
+        KwargInfo('method', str, default='auto'),
+        KwargInfo('preserve_paths', bool, default=False),
+    )
+    def qml_module(self, state: ModuleState, args: T.Tuple[str, str, str], kwargs: 'QmlModuleKwArgs') -> ModuleReturnValue:
+
+        self._detect_tools(state, kwargs['method'])
+        if not self._support_qml_module:
+            raise MesonException('qt.qml_module is not suppported for this version of Qt')
+
+        #Major.Minor(.Patch)
+        version_re = re.compile('^(\\d+)\\.(\\d+)(\\.(\\d+))?$')
+        output: T.List[T.Union[build.CustomTarget, build.GeneratedList]] = []
+
+        target_name: str = args[0]
+        module_name: str = args[1]
+        module_version: str = args[2]
+
+        module_version_match = version_re.match(module_version)
+        if not module_version_match:
+            raise MesonException(f'qml module version should be in the form Major.Minor, got {module_version}')
+        module_version_major: str = module_version_match.group(1)
+        module_version_minor: str = module_version_match.group(2)
+        #qt ignores .patch version
+
+        module_prefix_list: T.List[str] = module_name.split('.')
+        module_prefix: str = os.path.join(*module_prefix_list)
+        module_prefix_full: str = os.path.join(*(kwargs['resources_prefix'].split('/') + module_prefix_list))
+        module_path_outdir: str = os.path.join(state.environment.build_dir, kwargs['output_directory'] if kwargs['output_directory'] != '' else module_prefix)
+        os.makedirs(module_path_outdir, exist_ok=True)
+
+        qml_qresource_abs: str = os.path.join(module_path_outdir, 'qml.qrc')
+        qrc_resouces: T.List[T.Union[FileOrString, build.CustomTarget, build.CustomTargetIndex, build.GeneratedList]] = []
+        all_qml_files: T.Sequence[T.Union[FileOrString, build.CustomTarget]] = list(kwargs['qml_sources']) + list(kwargs['qml_singletons']) + list(kwargs['qml_internals'])
```

Comment:
I'm wondering if it would be beneficial to use the `Interpreter.source_strings_to_files` method here to remove the strings and just have `list[File | CustomTarget | CustomTargetIndex | GeneratedList]` here, and then not have to do string handling internally.